lib/pbench/server/api/__init__.py

```diff
@@ -130,6 +131,12 @@ def register_endpoints(api: Api, app: Flask, config: PbenchServerConfig):
         endpoint="endpoints",
         resource_class_args=(config,),
     )
+    api.add_resource(
```

Comment:
Our REST APIs are `noun[/verb]` form; we're effectively managing an API Key resource here, so `/key/generate` might work; however it's also reasonable that a `POST /key` would *create* a key resource, and that also makes sense to provide an explicit `DELETE` on the same endpoint... whereas `DELETE /generate_key` makes no sense.

(Arguably `/upload/{name}` should have been `PUT /datasets`, which would return the `resource_id` or even a `location: /api/v1/datasets/{resource_id}`, but let's at least not add another weird one.)